record/record.go

```diff
@@ -24,15 +24,30 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
-//				break
-//			}
 //			if err != nil {
-//				return nil, err
+//				if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
It'd be nice (and also not break any existing code) if we could preserve the property that the caller is meant to break only on io.EOF, and not also on io.ErrUnexpectedEOF. I think we can do that. (See below).

The example code then becomes:
rec, err := records.Next()
if err == io.EOF {
  break
}
if err != nil {
  log.Printf("recovering from %v", err)
  records.Recover()
  continue
}
// Ditto if you get an error during ioutil.ReadAll.